x-pack/plugins/security_solution/common/endpoint/schema/trusted_apps.test.ts

```diff
@@ -389,7 +401,23 @@ describe('When invoking Trusted Apps Schema', () => {
       value: 'c:/programs files/Anti-Virus',
       ...(data || {}),
     });
-    const createNewTrustedApp = <T>(data?: T): NewTrustedApp => ({
+    const createNewTrustedApp = <T>(
+      data?: T
+    ):
+      | (T & {
+          effectScope: { type: string };
+          entries: TrustedAppConditionEntry[];
+          os: OperatingSystem.WINDOWS;
+          name: string;
+          description: string;
+        })
+      | {
+          effectScope: { type: string };
+          entries: TrustedAppConditionEntry[];
+          os: OperatingSystem.WINDOWS;
+          name: string;
+          description: string;
+        } => ({
```

Comment:
i've been thinking about this change. i can't put my finger on it why there is a type error without this change, but it feels broken. i mean - we cannot assign an array of `TrustedAppConditionEntry` type to `NewTrustedApp['entries']`? it just doesn't seem right

i'll think about this a bit more. do you have any idea?

Reply:
looks like the problem roots around using enums as generics, where we want to create a type where only a partial set of the enum elements are should be used

when we create `TrustedAppConditionEntry`, we want its `field: T` to possibly have the values ot the `ConditionEntryField` enum, but in cases, only a subset of those (e.g. Linux only has path and hash)
```ts
export interface TrustedAppConditionEntry<T extends ConditionEntryField = ConditionEntryField>
```

but, in the test, we fall back to the default generic type, `ConditionEntryField`, that has all enum elements

so far it has worked, because accidentally one of the derived types contained **all elements** of the enum:
```ts
export type WindowsConditionEntry = TrustedAppConditionEntry<
  ConditionEntryField.HASH | ConditionEntryField.PATH | ConditionEntryField.SIGNER
>;
```
which ensured that there is a possibility, using the default of the generic (i.e. `ConditionEntryField`), to meet this criteria


but, with your modification, now neither of the derived types contains all of the enum elements, hence the error message. and i have no idea how to fix it either in the generic rule, or defining the generics type in the test file

still, looks like there's a nice quick fix to avoid the type mess: type assertion! i'll add it in a new comment


---

btw, i was worried about this issue, bc type system is one of our guard on code quality, and although automated and manual tests showed that everything works fine, still didn't want to throw out the typing

Reply:
couldn't add it in a suggestion, bc the lines are out of your modifications, so here it is simply:

```ts
    const createNewTrustedApp = <T>(data?: T): NewTrustedApp =>
      ({
        name: 'Some Anti-Virus App',
        description: 'this one is ok',
        os: OperatingSystem.WINDOWS,
        effectScope: { type: 'global' },
        entries: [createConditionEntry()],
        ...(data || {}),
      } as NewTrustedApp);
```

simply asserting the created object `as NewTrustedApp` solves the type error, while ensures that the created object aligns with the expectations in both places

Reply:
Makes total sense! Thanks for looking into this, I appreciate it! 9d393761cb0b106ecb541f98045e9480e6d6a3d3